mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala

```diff
@@ -417,6 +435,17 @@ object PowerIterationClustering extends Logging {
       .setK(k)
       .setSeed(0L)
       .run(points.values)
-    points.mapValues(p => model.predict(p)).cache()
+
+    val predict = points.mapValues(p => model.predict(p)).cache()
+    predict.count()
+    points.unpersist()
+    predict
```

Comment:
Why not just not cache `predict` here? that avoids dealing with unpersisting in the single caller above.

Reply:
Yes. removed the caching of `predict`